src/components/atoms/UserStatusDropdown/UserStatusDropdown.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { UserStatus } from "types/User";
+import { useShowHide } from "hooks/useShowHide";
+
+import "./UserStatusDropdown.scss";
+
+export interface UserStatusDropdownProps {
+  options: Array<UserStatus>;
+  onChange: (option: UserStatus) => void;
+  label: UserStatus | string;
+}
+
+export const UserStatusDropdown: React.FC<UserStatusDropdownProps> = ({
+  options,
+  label,
+  onChange,
+}) => {
+  const { isShown, hide, toggle } = useShowHide();
+
+  const onOptionClicked = (value: UserStatus) => () => {
+    onChange(value);
+    hide();
+  };
```

Comment:
useCallback + when you useMemo `optionComponents`, you can remove one layer of function and make it not factory () => () => void, but a simple function () => void

Reply:
https://github.com/sparkletown/sparkle/pull/1384/commits/2534af4e00cb14c31c12a45ed214f0d543c84981